scripts/post_model.js

```diff
@@ -0,0 +1,185 @@
+/* eslint-disable no-unused-vars */
+/**
+ * PostModel class
+ */
+class PostModel {
+  _posts = [];
+
+  _ids = new Set();
```

Comment:
Cosmetic (architectural): Hm interesting, basically you are near one of the nice solutions 
See: you need to have to store id's set and posts themselves, but you forgot one intersecting fact that makes things easy - id and post itself has a relation (id -> post). So that means we can have just one special collection here - Map.